### PR TITLE
[Storage] Refactor tests in `test_avro_async.py`

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This is the start of the work to help clear up some failures in the pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1972571&view=ms.vss-test-web.build-test-results-tab

This fixes the tests in the tests/avro path

Additionally, this fixes a typo (`test_with_hearder_reader` --> `test_with_header_reader`)

### CODE REVIEW

File: sdk/storage/azure-storage-blob/tests/avro/test_avro_async.py [hunk 1]
```diff
@@ -84,9 +84,8 @@ async def _test_reader(self):
             correct,
             len(CODECS_TO_VALIDATE) * len(SCHEMAS_TO_VALIDATE))
 
-    def test_reader(self):
-        loop = asyncio.get_event_loop()
-        loop.run_until_complete(self._test_reader())
+    async def test_reader(self):
+        await self._test_reader()
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/tests/avro/test_avro_async.py:88]: I would say get rid of these methods all together and just get rid of `_` in `_test_reader` to make that the test that is automatically run. No need for this helper method now :)

Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/tests/avro/test_avro_async.py:88]: Ah gotcha. I remember us chatting about this offline and was slightly confused but this clears it up 😄

**Review by jalauzon-msft — APPROVED** [review]
Thanks for jumping on this!

**Pull request merged**